[Thuya] iP

athuyaoo opened the pull request.

Commit:
Add Gradle support

athuyaoo committed:
Complete level 1

athuyaoo committed:
Complete level 2

athuyaoo committed:
Complete level 3

athuyaoo committed:
Complete Level 4

athuyaoo committed:
Add Automated Text Ui Testing

athuyaoo committed:
Add Level 5: Handle Errors

athuyaoo committed:
Add error handling and test cases

athuyaoo committed:
Add Level 6: Delete

athuyaoo committed:
Update tests

athuyaoo committed:
Add save

benclmnt commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+        System.out.println(wrapDivider(
+                "     Noted. I've removed this task: \n" +
+                "       " + task.toString() + "\n" +
+                "     Now you have "+ taskList.size() +" tasks in the list.\n"
```
I like how you separate this long string into different lines, but perhaps it is better to have breaks before operators?
```suggestion
                "     Noted. I've removed this task: \n"
                + "       " + task.toString() + "\n"
                + "     Now you have "+ taskList.size() +" tasks in the list.\n"
```

athuyaoo replied:
ok! thank you!

benclmnt commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+    private static int getTaskIndex(String[] commands, List<Task> taskList) throws DukeException {
+        if (commands.length < 2) {
+            throw new DukeException("☹ OOPS!!! Please put a number to select a task for the \"" +
+                    commands[0] + "\" action!");
```
Perhaps it is better to put break before operators? I noticed the same issue in several other places too

```suggestion
            throw new DukeException("☹ OOPS!!! Please put a number to select a task for the \""
                    + commands[0] + "\" action!");
```

athuyaoo replied:
Thanks! Will Fix

benclmnt commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,140 @@
+                    sayBye();
+                    return false;
+                default:
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
```
Perhaps not having any indentation for `case` clauses will help improve readability?

athuyaoo replied:
Thanks ! Will fix.

benclmnt commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,18 @@
+public abstract class Task {
+    protected boolean completed = false;
```
Perhaps the name could be changed to `isCompleted` to sound like a booleans?

athuyaoo replied:
Great suggestion! Thank you!

benclmnt reviewed (commented):
Overall, I found your code easy to read for the most part. I noted a few minor coding standard violations too, but overall it looks good to me!

athuyaoo committed:
Fix bugs, update code, and tests

athuyaoo committed:
Implement date for deadline

athuyaoo committed:
Merge branch 'branch-Level-7' into master

athuyaoo committed:
Merge branch 'branch-Level-8' into master

athuyaoo committed:
Fix bugs

athuyaoo committed:
Make the code more OOP

athuyaoo committed:
Add JUnit Tests

athuyaoo committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

athuyaoo committed:
Add Gradle to Project

athuyaoo committed:
Merge branch 'branch-A-Gradle' into master

LeeMingDe commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task  {
```
Hi, i think it would be nice to add java docs for this!

athuyaoo replied:
Added! Thanks!

LeeMingDe commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+public class Deadline extends Task  {
+    private LocalDate date;
+
+    Deadline(String name, String dateString) {
```
did you forget to add in an access modifier here?

athuyaoo replied:
Why do you say that? Do we need to do so?

LeeMingDe replied:
ah sorry my bad, its just my habit to add access modifiers

LeeMingDe commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+        this.date = LocalDate.parse(dateString);
+    }
+
+    public static Deadline create(String description) throws DukeException {
```
Can add in java docs here too!

athuyaoo replied:
Great will add!

LeeMingDe commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,80 @@
+import java.nio.file.*;
+import java.util.List;
+
 public class Duke {
```
Think you can add java docs here too!

athuyaoo replied:
Thanks! Will add.

LeeMingDe commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,29 @@
+public class Event extends Task  {
```
Noticed that java docs are missing for all the classes and methods, might be good to add it in!

athuyaoo replied:
Great! Will add.

LeeMingDe commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,64 @@
+public class Storage {
+    private Path savePath;
+
+    Storage(Path savePath) {
```
Access modifier missing here perhaps?

athuyaoo replied:
What access modifier do you recommend I add?

LeeMingDe commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,64 @@
+            return taskList;
+        }
+
+        try (Stream<String> stream = Files.lines(savePath)) {
```
Nice use of streams, never thought about it that way, good job!

athuyaoo replied:
Thanks!

LeeMingDe reviewed (commented):
Good job overall! Your code is clean and easy to read. However, you did not add in java docs for all your classes and methods. I think it will improve readability if you add those in as well!

athuyaoo committed:
Add javadoc comments to the code

athuyaoo committed:
Follow coding standard

athuyaoo committed:
Add find command

As per level-9 request, Give users a way to find a task by searching for
a keyword.

athuyaoo committed:
Add javadoc comment

athuyaoo committed:
Merge branch 'branch-A-JavaDoc' into master

athuyaoo committed:
Merge branch 'branch-A-CodingStandard' into master

athuyaoo committed:
Merge branch 'branch-Level-9' into master

athuyaoo committed:
Fix coding standard issues

athuyaoo committed:
Add Gui to Duke

athuyaoo committed:
Merge branch 'branch-Level-10' into master

athuyaoo committed:
Add checkstyle and fix errors

athuyaoo committed:
Merge branch 'branch-A-CheckStyle' into master

athuyaoo committed:
Change main class to launcher

athuyaoo committed:
Fix TaskListTest

athuyaoo committed:
Use assertions

athuyaoo committed:
Improve Code Quality to Task Class

athuyaoo committed:
Improve code quality of parser class

athuyaoo committed:
Improve code quality of Deadline Event and Parser class

athuyaoo committed:
Fix bug where list of 0 doesn't show

athuyaoo committed:
Merge pull request #1 from athuyaoo/branch-A-Assertions

Use assertions

athuyaoo committed:
Merge branch 'master' into branch-A-CodeQuality

athuyaoo committed:
Fix find command bug in TaskList class

athuyaoo committed:
Improve checkstyle

athuyaoo committed:
Merge branch 'master' into branch-A-CodeQuality

athuyaoo committed:
Merge pull request #2 from athuyaoo/branch-A-CodeQuality

Merge branch-A-CodeQuality

athuyaoo committed:
Add flexibile data source

athuyaoo committed:
Merge branch 'branch-B-FlexibleDataSource' into master

athuyaoo committed:
Add minor textual changes

athuyaoo committed:
Add icon and change text of window

athuyaoo committed:
Fix junit tests

athuyaoo committed:
Add user guide

athuyaoo committed:
Update user guide
